atsign-foundation/at_server
# feat: <breaking_change> persistence spec refactoring to replace metadata params with metadata object from at_commons

**- What I did**
- refactored persistence spec to replace optional metadata params with metadata from at_commons
- any future addition of metadata will not affect the method signature in the spec
**- How I did it**
- added dependency to at_commons
- changed method signature of put/create methods in Keystore to replace optional metadata params with metadata object from at_commons 
- removed putAll method since put method should suffice for updating either value or metadata or both
**- How to verify it**
- verify the tests in persistence secondary and server after uptaking the spec changes
- tests in this PR
https://github.com/atsign-foundation/at_server/pull/1822
